### PR TITLE
feat: Enable monitoring via Prometheus and Grafana (rebased)

### PR DESCRIPTION
**Pull request by philippemnoel**
# Ticket(s) Closed

- Closes #N/A

## What
Rebased version of #46 

CNPG supports monitoring via Prometheus and Grafana, but we haven't enabled it yet. This PR fixes that. I followed the following two links, shared by @itay-grudev:
- https://cloudnative-pg.io/documentation/current/monitoring/
- https://cloudnative-pg.io/documentation/current/quickstart/#part-4-monitor-clusters-with-prometheus-and-grafana

TODOs:
- [x] Update the CI to log Prometheus/Grafana pods for visibility
- [x] Configure TLS over the metrics -- EDIT: Configured by default
- [x] Evaluate whether to configure any user-defined metrics -- EDIT: This is in Beta. Let's wait for customers to request it, if any, and that feature will have progressed as well.
- [x] Write documentation for Prometheus
- [x] Update documentation to say that unless they already have a Prometheus deployment
- [x] Update the documentation to create a namespace in Prometheus (?)
- [x] Test in CI
- [x] Test locally in Minikube
- [x] Do I need to create a PrometheusRule? -- EDIT: Not needed since Prometheus rules are preconfigured in the cluster
- [x] ~Add the pre-configured Grafana dashboards (the plain install doesn't ship with any dashboard) -- https://github.com/cloudnative-pg/grafana-dashboards, https://github.com/cloudnative-pg/charts/blob/c7cb7be3aa24a72f1d10f28499cf83a7963b6b5b/charts/cloudnative-pg/values.yaml#L161~ EDIT: It is automatically packaged in the Helm chart
- [x] Write documentation for Grafana
- [x] Final Minikube test with all the latest written documentation
- [x] Submit a PR to edit our docs in `paradedb/paradedb` -- https://github.com/paradedb/paradedb/pull/1784

## Why
So we can have visibility when we do BYOC.

## How
Followed the links above

## Tests
It all works, and I opened a new issue to further improve the dashboard

### CODE REVIEW

File: .github/workflows/paradedb-test-eks.yml [hunk 1]
```diff
@@ -88,11 +88,32 @@ jobs:
...
+        run: |
+          kubectl wait --namespace prometheus-community --for=condition=available --timeout=120s deployment/prometheus-community-grafana
+          kubectl wait --namespace prometheus-community --for=condition=available --timeout=120s deployment/prometheus-community-kube-operator
+          kubectl wait --namespace prometheus-community --for=condition=available --timeout=120s deployment/prometheus-community-kube-state-metrics
+          kubectl get svc -n prometheus-community prometheus-community-grafana
+          kubectl get svc -n prometheus-community prometheus-community-kube-alertmanager
+          kubectl get svc -n prometheus-community prometheus-community-kube-operator
+          kubectl get svc -n prometheus-community prometheus-community-kube-prometheus
+          kubectl get svc -n prometheus-community prometheus-community-kube-state-metrics
+          kubectl get svc -n prometheus-community prometheus-operated
```
Comment (itay-grudev) [inline .github/workflows/paradedb-test-eks.yml:115]: For the purposes of this test is installing the whole Prometheus stack necessary? It's quite heavy. Installing just the CRDs to enable testing without any UIs or actual metric collection is sufficient.

```
helm upgrade --install --atomic --namespace prometheus-community prometheus-crds prometheus-community/prometheus-operator-crds
```

Comment (itay-grudev) [reply .github/workflows/paradedb-test-eks.yml:115]: In my the example above `--atomic` will make sure the chart is installed and will wait, so there won't be any need for wait step and it should be removed.

Comment (philippemnoel) [reply .github/workflows/paradedb-test-eks.yml:115]: Oh no way

Comment (philippemnoel) [reply .github/workflows/paradedb-test-eks.yml]: Done

**Pull request merged**